querqy-for-lucene/querqy-lucene/src/main/java/querqy/lucene/rewrite/BoostedDelegatingFieldBoost.java

```diff
@@ -0,0 +1,36 @@
+package querqy.lucene.rewrite;
+
+import java.io.IOException;
+
+import org.apache.lucene.index.IndexReader;
+
+import querqy.model.Term;
+
+public class BoostedDelegatingFieldBoost implements FieldBoost {
+
+    private final FieldBoost delegate;
+    private final float boost;
+
+    public BoostedDelegatingFieldBoost(final FieldBoost delegate, final float boost) {
+        this.delegate = delegate;
+        this.boost = boost;
+    }
+
+    @Override
+    public float getBoost(final String fieldname, final IndexReader indexReader) throws IOException {
+        final float computedBoost = delegate.getBoost(fieldname, indexReader);
+        return computedBoost * boost;
+    }
+
+    @Override
+    public void registerTermSubQuery(final String fieldname, final TermSubQueryFactory termSubQueryFactory, final Term sourceTerm) {
+        delegate.registerTermSubQuery(fieldname, termSubQueryFactory, sourceTerm);
+    }
+
+    @Override
+    public String toString(final String fieldname) {
+        return delegate.toString(fieldname);
```

Comment:
We should let the outer world know that a boost was applied and add the boost to the string. Same for hashcode()/equals()

Reply:
Implemented in 14d046a